File: python/128_Longest_Consecutive_Sequence.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def longestConsecutive(self, nums):
        if not nums or len(nums)<2:
            return len(nums)
        p,max={},1
        for num in nums:
            p[num]=1
        for num in nums:
            p[num]=0
            k,cnt=num+1,1
            while p.get(k,0)!=0:
                p[k]=0
                cnt,k=cnt+1,k+1
            k=num-1
            while p.get(k,0)!=0:
                p[k]=0
                cnt,k=cnt+1,k-1
            max=cnt if max<cnt else max
        return max

File: Java/128_Longest_Consecutive_Sequence.java
public class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums==null)return 0;
        if(nums.length<=1)return nums.length;
        HashMap<Integer, Boolean> map = new HashMap<Integer, Boolean>();
        for(int num:nums){
            map.put(num, true);
        }
        int res=1, cur,k, len=nums.length;
        for(int i=0;i<len;i++){
            if(!map.get(nums[i]))continue;
            cur=1;
            k=1;
            while(map.containsKey(nums[i]+k)){
                map.put(nums[i]+k, false);
                cur++;
                k++;
            }
            k=1;
            while(map.containsKey(nums[i]-k)){
                map.put(nums[i]-k, false);
                cur++;
                k++;
            }
            res=Math.max(res,cur);
        }
        return res;
    }
}